src/Nethermind/Nethermind.Db.Rocks/DbOnTheRocks.cs

```diff
@@ -805,11 +811,37 @@ public void Dispose()
             }
             _isDisposed = true;
 
+            Dictionary<ValueKeccak, (byte[]? data, ColumnFamilyHandle? cf)> batchData = _batchData;
```

Comment:
Why not `SortedDictionary<>`?

Reply:
Looks like it allocates a new `Node` on each item inserted (linked list) so doesn't get any benefit of Dictionary reuse. The Linq sort will also allocate, but is very short lived so Gen0 clean up